### PR TITLE
CI: Create Automated Workflow for Translation Catalog Updates

### LINKED ISSUE #5961: Make Translation Catalog More in Sync with Source
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

Translation catalogs are often updated right before a (pre-)release and the translations for the new features end up not being available in the build.

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

Let translation catalog be updated more frequently.

### Describe alternatives you've considered

Update by hand every once in a while (not desirable).

### PR DESCRIPTION
**Pull request by ElectricalBoy**
# Description

This PR adds an automation workflow that updates the translation catalog when the relevant source files are updated.

The workflow is triggered by changes of the file(s) listed in [localization/i18n/list.txt](https://github.com/SoftFever/OrcaSlicer/blob/main/localization/i18n/list.txt).

Resolves #5961

## Tests

- Proof-of-concept workflow: <https://github.com/ElectricalBoy/OrcaSlicer/actions/runs/9737476435>
  Generated d7f6dc8

### CODE REVIEW

File: .github/workflows/update-translation.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: Update Translation Catalog
+on: 
+    push:
+      branches: 
+       - main
+      paths:
+        - src/**
```
Comment (Ocraftyone) [inline .github/workflows/update-translation.yml:7]: Maybe specify libslic3r and slic3r with files that end in .cpp/.hpp? Could save a few unnecessary runs.

**Comment by SoftFever** [comment]
I'm worried that we would be having too many auto locale commits that will make the git history basically unreadable :(

**Comment by SoftFever** [comment]
> > I'm worried that we would be having too many auto locale commits that will make the git history basically unreadable :(
> 
> What if we set this workflow to have a scheduled trigger (e.g., once a week)?
> 
> Ref: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule

Hmm, can we make it manual?

**Comment by Ocraftyone** [comment]
> Btw is there a way to build [HintsToPot.cpp](https://github.com/SoftFever/OrcaSlicer/blob/main/src/hints/HintsToPot.cpp?rgh-link-date=2024-07-02T01%3A54%3A31Z) only without building the entire slicer?

yes. you run the cmake configure command, then you would build with the target of hintsToPot

**Comment by SoftFever** [comment]
> My intent is to remove the "human interaction required" part with this PR, and I think that manual trigger only doesn't really solve the problem (as stated in #5961). If the history clogging up is an issue (which I think is a valid concern), how about:
> 
> * Making the workflow interval to something less aggressive (e.g., bi-weekly), or
> * Making the workflow to create commits directly to the PRs so that everything can get squashed before merging, or
> * Making GHA to create PRs (and update existing PR until merged) instead of making direct commits?

Hmm, I'm not sure.
To be honest, I would rather run them manually.
I pay close attention to decide when to update the locale as well as the merging order to cause less hassle for translation authors.
That's also the original intention behind isolating the locale build system. Many translators are not familiar with Git.

**Review by SoftFever — APPROVED** [review]
Looks good
Thank you!

**Pull request merged**